File: examples/asyncapi-shopping-cart/shopping-cart/src/test/java/io/example/asyncapi/shoppingcart/config/ServicesInMemoryConfig.java
package io.example.asyncapi.shoppingcart.config;

import io.example.asyncapi.shoppingcart.*;
import io.example.asyncapi.shoppingcart.domain.*;
import io.example.asyncapi.shoppingcart.events.*;
import java.util.List;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

/**
 * Services InMemory Config. It can be used standalone or with @SpringBootTest.
 */
@Configuration
@Profile("in-memory")
public class ServicesInMemoryConfig extends RepositoriesInMemoryConfig {

    protected final EventsProducerInMemoryContext eventsProducerInMemoryContext = new EventsProducerInMemoryContext();

    protected final ShoppingCartServiceImpl shoppingCartService = new ShoppingCartServiceImpl(
            shoppingCartRepository(), eventsProducerInMemoryContext.shoppingCartEventsProducer());

    @Bean
    public <T extends ShoppingCartService> T shoppingCartService() {
        return (T) shoppingCartService;
    }

    static List<ShoppingCart> _shoppingCarts;

    public void reloadTestData() {

        var testDataLoader = new TestDataLoader(List.of(ShoppingCart.class, Item.class));
        var shoppingCarts = _shoppingCarts != null
                ? _shoppingCarts
                : testDataLoader.loadCollectionTestDataAsObjects(ShoppingCart.class);
        shoppingCartRepository().deleteAll();
        shoppingCartRepository().saveAll(shoppingCarts);
    }

    public EventsProducerInMemoryContext getEventsProducerInMemoryContext() {
        return eventsProducerInMemoryContext;
    }
}

File: examples/order-fulfillment-kotlin/src/test/kotlin/io/zenwave360/example/orderfulfillment/config/ServicesInMemoryConfig.kt
package io.zenwave360.example.orderfulfillment.config

import io.zenwave360.example.orderfulfillment.*
import io.zenwave360.example.orderfulfillment.domain.*
import io.zenwave360.example.orderfulfillment.events.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

/** Services InMemory Config. It can be used standalone or with @SpringBootTest. */
@Configuration
@Profile("in-memory")
open class ServicesInMemoryConfig : RepositoriesInMemoryConfig() {

    protected val eventsProducerInMemoryContext = EventsProducerInMemoryContext()

    protected val orderService =
        OrderServiceImpl(orderRepository(), eventsProducerInMemoryContext.orderEventsProducer())

    @Bean
    open fun <T : OrderService> orderService(): T {
        return orderService as T
    }

    companion object {
        var _orders: List<Order>? = null
    }

    fun reloadTestData() {

        val testDataLoader = TestDataLoader(listOf(Order::class.java))
        val orders = _orders ?: testDataLoader.loadCollectionTestDataAsObjects(Order::class.java)
        orderRepository().deleteAll()
        orderRepository().saveAll(orders)
    }
    // fun getEventsProducerInMemoryContext(): EventsProducerInMemoryContext {
    //     return eventsProducerInMemoryContext
    // }
}
